model/converter/json/from_domain.go

```diff
@@ -126,29 +153,57 @@ func (fd fromDomain) convertKeyValues(keyValues model.KeyValues) []json.KeyValue
 	return out
 }
 
-func (fd fromDomain) convertLogs(logs []model.Log) []json.Log {
+func (fd fromDomain) convertESKeyValues(keyValues model.KeyValues) []json.KeyValue {
+	out := make([]json.KeyValue, len(keyValues))
+	for i, kv := range keyValues {
+		out[i] = json.KeyValue{
+			Key:   kv.Key,
+			Type:  json.ValueType(kv.VType.String()),
+			Value: kv.AsString(),
+		}
+	}
+	return out
+}
+
+func (fd fromDomain) convertLogs(logs []model.Log, es bool) []json.Log {
 	out := make([]json.Log, len(logs))
 	for i, log := range logs {
+		var fields []json.KeyValue
+		var timestamp uint64
+		if es {
+			fields = fd.convertESKeyValues(log.Fields)
+			timestamp = uint64(log.Timestamp.UnixNano())
```

Comment:
why does this need to be different from non-ES version? The default precision for timestamps is microseconds, there's no benefit in converting to nanos.